modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/PublicationChannelToWorkspace.java

```diff
@@ -90,67 +89,62 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
 
     Optional<Publication> publication = Arrays.stream(mediaPackage.getPublications())
         .filter(channel -> channel.getChannel().equals(publicationChannel)).findFirst();
-    if (publication.get().getTracks().length >= 0) {
-
-      Collection<MediaPackageElement> tracks =  new ArrayList<MediaPackageElement>();
-
-      //get tracks from publicationchannel with tags and flavors
-      Arrays.stream(publication.get().getTracks())
-          .filter(element -> element.containsTag(configuredSourceTags))
-          .filter(Objects::nonNull)
-          .forEach(element -> tracks.add(element));
-      Arrays.stream(publication.get().getTracks())
-          .filter(element -> configuredSourceFlavors.contains(element.getFlavor()))
-          .filter(Objects::nonNull)
-          .forEach(element -> tracks.add(element));
-
-      if (!configuredTargetTags.isEmpty()) {
-        tracks.stream().forEach(track -> {
-          configuredTargetTags.stream().forEach(targetTag -> track.addTag(targetTag.toString()));
-        });
-      }
-      tracks.stream().forEach(track -> mediaPackage.add(track));
+
+    // Skip if the media package does not contain a matching publication
+    if (publication.isEmpty()) {
+      logger.debug("The media package '{}' contains no publications on channel '{}', skipping.",
+          mediaPackage.getIdentifier(), publicationChannel);
```

Comment:
```suggestion
      logger.warn("The media package '{}' contains no publications on channel '{}', skipping.",
          mediaPackage.getIdentifier(), publicationChannel);
```

I like this more  `warn` than `debug`, because if there is an error on the workflow, I want to be warned that the MP doesn't contain the media to be published.